pandas/core/groupby/generic.py

```diff
@@ -693,14 +693,14 @@ def value_counts(
         # for compat. with libgroupby.value_counts need to ensure every
         # bin is present at every index level, null filled with zeros
         diff = np.zeros(len(out), dtype="bool")
-        for lab in labels[:-1]:
-            diff |= np.r_[True, lab[1:] != lab[:-1]]
+        for codes_ in codes[:-1]:
```

Comment:
code?  val?

Reply:
Codes is a list of arrays here, so I can call it level_codes.